apps/web/app/(app)/environments/[environmentId]/settings/lookandfeel/components/UnifiedStylingPreviewSurvey.tsx

```diff
@@ -0,0 +1,255 @@
+                    onClick={() => {
+                      setshrink(true);
+                      setPreviewPosition("relative");
+                      setTimeout(() => setIsFullScreenPreview(false), 300);
+                    }}
+                  />
+                ) : (
+                  <ArrowsPointingOutIcon
+                    className="mr-2 h-4 w-4 cursor-pointer"
+                    onClick={() => {
+                      setshrink(false);
+                      setIsFullScreenPreview(true);
+                      setTimeout(() => setPreviewPosition("fixed"), 300);
+                    }}
+                  />
+                )}
+                <ResetProgressButton resetQuestionProgress={resetQuestionProgress} />
+              </div>
+            </div>
+          </div>
+
+          {previewType === "web" ? (
+            <Modal
+              isOpen
+              placement={placement}
+              highlightBorderColor={highlightBorderColor}
+              previewMode="desktop"
+              background={styling.cardBackgroundColor?.light}
+              borderRadius={styling.roundness ?? 12}>
+              <SurveyInline
+                survey={survey}
+                activeQuestionId={activeQuestionId || undefined}
+                isBrandingEnabled={product.inAppSurveyBranding}
+                onActiveQuestionChange={setActiveQuestionId}
+                isRedirectDisabled={true}
+                onFileUpload={onFileUpload}
+                styling={styling}
+              />
+            </Modal>
+          ) : (
+            <MediaBackground survey={survey} ContentRef={ContentRef} isEditorView>
+              <div className="z-0 w-full max-w-md rounded-lg p-4">
+                <SurveyInline
+                  survey={survey}
+                  activeQuestionId={activeQuestionId || undefined}
+                  isBrandingEnabled={product.linkSurveyBranding}
+                  onActiveQuestionChange={setActiveQuestionId}
+                  isRedirectDisabled={true}
+                  onFileUpload={onFileUpload}
+                  responseCount={42}
+                  styling={styling}
+                />
+              </div>
+            </MediaBackground>
+          )}
+        </div>
+      </motion.div>
+
+      {/* for toggling between mobile and desktop mode  */}
+      <div className="mt-2 flex rounded-full border-2 border-slate-300 p-1">
+        <div
+          className={`${previewType === "link" ? "rounded-full bg-slate-200" : ""} cursor-pointer px-3 py-1 text-sm`}
+          onClick={() => setPreviewType("link")}>
+          Link survey
+        </div>
+
+        <div
+          className={`${previewType === "web" ? "rounded-full bg-slate-200" : ""} cursor-pointer px-3 py-1 text-sm`}
+          onClick={() => setPreviewType("web")}>
+          App survey
+        </div>
+      </div>
+    </div>
+  );
+}
+
+function ResetProgressButton({ resetQuestionProgress }) {
```

Comment:
I always prefer to only have one component per file to increase readability and since the file is already pretty complex and big, let's move this to it's own file to make it easier to navigate.

Also please use `const ResetProgressButton = ` syntax to be consistent and also type the props.